src/consumerIncentives/saga.ts

```diff
@@ -230,8 +234,18 @@ export function* fetchAvailableRewardsSaga() {
       )
     }
 
-    yield put(setAvailableRewards(data.availableRewards))
-    yield put(fetchAvailableRewardsSuccess())
+    // as there is a race condition on app start between fetching the
+    // supercharge rewards and fetching remote config values, prevent mismatched
+    // data from being stored in redux here
+    const mostRecentSuperchargeV2Enabled = yield select(superchargeV2EnabledSelector)
+    if (
+      data.availableRewards.length === 0 ||
+      (mostRecentSuperchargeV2Enabled && isSuperchargePendingRewardsV2(data.availableRewards)) ||
+      (!mostRecentSuperchargeV2Enabled && !isSuperchargePendingRewardsV2(data.availableRewards))
+    ) {
+      yield put(setAvailableRewards(data.availableRewards))
+      yield put(fetchAvailableRewardsSuccess())
```

Comment:
Oh no, a double dispatch 😢  No need to do anything though, most of the code here do that.